libs/pdi-scanner/src/rust/client.rs

```diff
@@ -73,6 +74,28 @@ impl<T> Client<T> {
         }
     }
 
+    // There should only be one command/response occurring at a time, so before
+    // we send a command that expects a response, we should clear out any old
+    // responses. This ensures we don't get an outdated response (e.g. a previous scanner
+    // status that didn't get received when its command was sent). Thus, this
+    // method is called in the send_and_recv! macro.
+    //
+    // This accounts for the fact that there are cases where the scanner will
+    // delay sending a response to a command (e.g. when the "eject pause"
+    // feature pauses the scanner, it will queue commands received during that
+    // time).
+    fn clear_unhandled_packets_except_events(&mut self) {
+        let unhandled_non_event_packets = self
+            .unhandled_packets
+            .iter()
+            .filter(|packet| !packet.is_event())
+            .collect::<Vec<_>>();
+        if unhandled_non_event_packets.len() > 0 {
```

Comment:
Per clippy:

```suggestion
        if !unhandled_non_event_packets.is_empty() {
```

Reply:
thanks, changed.

maybe we should run clippy automatically?